DirectProgramming/DPC++FPGA/ReferenceDesigns/qri/src/qri_demo.cpp

```diff
@@ -0,0 +1,602 @@
+#include <math.h>
+
+#include <CL/sycl.hpp>
+#include <chrono>
+#include <iomanip>
+#include <list>
+#include <sycl/ext/intel/ac_types/ac_complex.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities//include/dpc_common.hpp
+#include "dpc_common.hpp"
+#include "qri.hpp"
+
+/*
+  COMPLEX, COLS_COMPONENT, ROWS_COMPONENT, FIXED_ITERATIONS_QRD and
+  FIXED_ITERATIONS_QRI are defined by the build system.
+  Depending on the value of COMPLEX, the real or complex QR based matrix
+  inversion function (QRI) is defined.
+
+  Each matrix (input and output) are represented using vectors and are
+  interpreted in a column fashion (transposed).
+
+  Function arguments:
+  - AMatrix:    The input matrix to be inverted.
+                Interpreted as a transposed matrix.
+  - invMatrix:  The output matrix. The function will overwrite this matrix.
+                Will contain the inverse of AMatrix.
+  - q:          The device queue.
+  - matrices:   The number of matrices to be processed.
+                The input matrices are read sequentially from the AMatrix
+                vector.
+  - reps:       The number of repetitions of the computation to execute.
+                (for performance evaluation)
+*/
+#if COMPLEX == 0
+// Real single precision floating-point QR based inversion
+void QRI(std::vector<float> &AMatrix, std::vector<float> &invMatrix,
+         sycl::queue &q, size_t matrices, size_t reps) {
+  constexpr bool isComplex = false;
+  QRI_impl<COLS_COMPONENT, ROWS_COMPONENT, FIXED_ITERATIONS_QRD,
+           FIXED_ITERATIONS_QRI, isComplex, float>(AMatrix, invMatrix, q,
+                                                   matrices, reps);
+}
+#else
+// Complex single precision floating-point QR based inversion
+void QRI(std::vector<ac_complex<float> > &AMatrix,
+         std::vector<ac_complex<float> > &invMatrix, sycl::queue &q,
+         size_t matrices, size_t reps) {
+  constexpr bool isComplex = true;
+  QRI_impl<COLS_COMPONENT, ROWS_COMPONENT, FIXED_ITERATIONS_QRD,
+           FIXED_ITERATIONS_QRI, isComplex, float>(AMatrix, invMatrix, q,
+                                                   matrices, reps);
+}
+#endif
+
+/*
+  Returns a random floating-point value between min and max
+*/
+float randomValueInInterval(float min, float max) {
+  return min + static_cast<float>(rand()) /
+                   (static_cast<float>(RAND_MAX) / (max - min));
+}
```

Comment:
Function names should be `CamelCase` with leading capital. As per samples coding style.

Reply:
Apply to all functions. Sorry :(